src/cpp/ext/otel/otel_plugin.h

```diff
@@ -384,6 +407,31 @@ class OpenTelemetryPlugin : public grpc_core::StatsPlugin {
     return plugin_options_;
   }
 
+  template <typename ValueType>
+  struct ObservableState {
+    void Observe(opentelemetry::metrics::ObserverResult& result)
+        ABSL_LOCKS_EXCLUDED(mu);
+
+    grpc_core::GlobalInstrumentsRegistry::InstrumentID id;
+    opentelemetry::nostd::shared_ptr<
+        opentelemetry::metrics::ObservableInstrument>
+        instrument;
+    bool callback_registered = false;
+    // We only support one Observable to one RegisteredMetricCallback and one
+    // RegisteredMetricCallback to multiple Observables relationship for now.
+    grpc_core::RegisteredMetricCallback* registered_metric_callback = nullptr;
+    // Per-instrument mutex to synchronize accesses to caches.
+    grpc_core::Mutex mu ABSL_ACQUIRED_AFTER(OpenTelemetryPlugin::mu_);
+    // Caches.
+    ValueType value ABSL_GUARDED_BY(mu);
+    std::vector<absl::string_view> label_values ABSL_GUARDED_BY(mu);
```

Comment:
Note that if we do need to copy the strings, then it may make sense to apply the optional label bitmask when we populate the metric data instead of when we put the data into OT, so that we don't copy label values that we don't actually need.  That would also allow us to store a single vector instead of 2, which would avoid another allocation.